### DISCUSSION TITLE
wrong final block length error

### DISCUSSION THREAD

**Discussion by sagarvision1406** [openssl/openssl#25685] [Q&A]
HI

I am using the below commands to encrypt and the decrypt the file in Unix and as well as in git bash, they are working fine as expected.

for encryption:- openssl enc -aes-256-cbc -salt -pbkdf2 -in mydata.txt -out mydata.txt.enc -pass pass:mysecretpassword
for decryption:- openssl enc -aes-256-cbc -d -salt -pbkdf2 -in mydata.txt.enc -out mydata.txt -pass pass:mysecretpassword

I have downloaded the above encrypt file into windows and trying to decrypt it by using the below command from git bash which is resulting the error "error:1c80006b: provident routines:ossl_cipher_generic_block_final:wrong final block length"

I am using the same command for decryption in the git bash as well which I have used in Unix.

Kindly help in sorting out this issue.

Regards,
Sagar

**Comment by mattcaswell** [comment]
Please first check that the file is identical and not become corrupted during the transfer.

E.g. run

`openssl dgst mydata.txt.enc` on both the Unix version of the file, and the windows version of the file and confirm that you get the same digest back on both machines.
